app/apps/server/storage/ConfigurableAppSourceStorage.ts

```diff
@@ -0,0 +1,54 @@
+import { AppSourceStorage, IAppStorageItem } from '@rocket.chat/apps-engine/server/storage';
+
+import { AppFileSystemSourceStorage } from './AppFileSystemSourceStorage';
+import { AppGridFSSourceStorage } from './AppGridFSSourceStorage';
+
+export class ConfigurableAppSourceStorage extends AppSourceStorage {
+	private filesystem: AppFileSystemSourceStorage;
+
+	private gridfs: AppGridFSSourceStorage;
+
+	private storage: AppSourceStorage;
+
+	constructor(storageType: string, filesystemStoragePath?: string) {
+		super();
+
+		this.filesystem = new AppFileSystemSourceStorage();
+		this.gridfs = new AppGridFSSourceStorage();
+
+		this.setStorage(storageType, filesystemStoragePath);
+	}
+
+	public setStorage(type: string, filesystemStoragePath?: string): void {
+		switch (type) {
+			case 'filesystem':
+				this.filesystem.setPath(filesystemStoragePath as string);
```

Comment:
I think we should split this into two methods; one for controlling the storage and another for controlling the `fileSystemStoragePath` - this makes it easier for us to just change one setting at a time, without needing the other.

Maybe we could even make the `this.filesystem` property publicly accessible, then change the path configuration directly on the setting callback

Reply:
Done! It works well